source/vulkan/vulkan_hooks.cpp

```diff
@@ -454,14 +693,399 @@ VkResult VKAPI_CALL vkQueuePresentKHR(VkQueue queue, const VkPresentInfoKHR *pPr
+{
+	PFN_vkQueueSubmit trampoline;
+
+	{ const std::lock_guard<std::mutex> lock(s_mutex);
+		trampoline = s_device_dispatch.at(get_dispatch_key(queue)).QueueSubmit;
+		assert(trampoline != nullptr);
+		assert(pSubmits != nullptr);
+
+		for (uint32_t i = 0; i < submitCount; ++i)
+		{
+			VkSubmitInfo pSubmit = pSubmits[i];
+			for (uint32_t j = 0; j < pSubmit.commandBufferCount; ++j)
+			{
+				VkCommandBuffer commandBuffer = pSubmits->pCommandBuffers[j];
+				const auto it = s_trackers_per_command_buffer.find(commandBuffer);
+				const auto it2 = s_queue_mapping.find(queue);
+				if (it != s_trackers_per_command_buffer.end() && it2 != s_queue_mapping.end())
+					merge_command_buffer_trackers(commandBuffer, s_device_tracker.at(it2->second));
+			}
+		}
+	}
+
+	return trampoline(queue, submitCount, pSubmits, fence);
+}
+
+VkResult VKAPI_CALL vkCreateImage(VkDevice device, const VkImageCreateInfo* pCreateInfo, const VkAllocationCallbacks* pAllocator, VkImage* pImage)
+{
+	PFN_vkCreateImage trampoline = nullptr;
+
+	{ const std::lock_guard<std::mutex> lock(s_mutex);
+		trampoline = s_device_dispatch.at(get_dispatch_key(device)).CreateImage;
+		assert(trampoline != nullptr);
+	}
+
+	VkImageCreateInfo createInfo = *pCreateInfo;
+
+	// check if it is a depth stencil buffer
+	if (pCreateInfo->usage & VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT)
+		createInfo.usage |= VK_IMAGE_USAGE_SAMPLED_BIT;
+
+	const VkResult result = trampoline(device, &createInfo, pAllocator, pImage);
+	if (result != VK_SUCCESS)
+	{
+		LOG(WARN) << "> vkCreateImage failed with error code " << result << '!';
+		return result;
+	}
+
+	// Guard access to the map
+	const std::lock_guard<std::mutex> lock(s_mutex);
+
+	// check if it is a depth stencil buffer
+	if (pCreateInfo->usage & VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT && pCreateInfo->usage & VK_IMAGE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT)
+		s_depth_stencil_buffer_images[*pImage] = { *pCreateInfo, device };
+
+	return result;
+}
+
+VkResult VKAPI_CALL vkCreateImageView(VkDevice device, const VkImageViewCreateInfo* pCreateInfo, const VkAllocationCallbacks* pAllocator, VkImageView* pView)
+{
+	PFN_vkCreateImageView trampoline = nullptr;
+
+	{ const std::lock_guard<std::mutex> lock(s_mutex);
+		trampoline = s_device_dispatch.at(get_dispatch_key(device)).CreateImageView;
+		assert(trampoline != nullptr);
+	}
+	
+	// create the original image view
+	VkResult result = trampoline(device, pCreateInfo, pAllocator, pView);
+	if (result != VK_SUCCESS)
+	{
+		LOG(WARN) << "> vkCreateImageView failed with error code " << result << '!';
+		return result;
+	}
+
+#if RESHADE_VULKAN_CAPTURE_DEPTH_BUFFERS
+	// create an image view where depth buffer is displayable in the shaders
+	VkImageViewCreateInfo createInfo = *pCreateInfo;
+	createInfo.subresourceRange.aspectMask = VK_IMAGE_ASPECT_DEPTH_BIT;
+
+	VkImageView pViewReplacement = VK_NULL_HANDLE;
+	result = trampoline(device, &createInfo, nullptr, &pViewReplacement);
```

Comment:
It's probably better to avoid creating a second image view for every image view the game creates. Maybe check whether the format in `pCreateInfo` is a depth format and then modify the aspect mask to always contain `VK_IMAGE_ASPECT_DEPTH_BIT` (before calling the trampoline to create the original image view). That way there is only one image view, which should support what we want it to do.

Reply:
Just tried it in Doom, but it appears that updating an actual image does not give to me the depth buffer, wherea the actuel code i made provide a valid one.